python/tank/descriptor/io_descriptor/base.py

```diff
@@ -320,14 +323,16 @@ def _get_locally_cached_versions(self):
                 for version_folder in os.listdir(parent_folder):
                     version_full_path = os.path.join(parent_folder, version_folder)
                     # check that it's a folder and not a system folder
-                    if os.path.isdir(version_full_path) and \
+                    if not self._is_settings_folder(version_full_path) and \
```

Comment:
You mean if you roll back the version of core? or if you have old and new versions of core using the same bundle cache concurrently? (it's a possibility).